File: Languages/Python/CodingChallenges/LeetCode/RemoveDuplicatesFromSortedArray/Attempt01.py
from typing import List

nums = [1, 1, 2]; # want [1, 2, 2]
# nums = [1, 1, 2, 3]; # want [1, 2, 3, 3]
# nums = [0, 0, 1, 1, 1, 2, 2, 3, 3, 4];

def removeDuplicates(nums: List[int]) -> int:# omitted "self" from signature.
# def removeDuplicates(self, nums: List[int]) -> int:
    if(len(nums)<2):
        return len(nums)
    i = 0
    j = 1
    while(j < len(nums)):
        if(nums[i]==nums[j]):
            j+= 1
        else:
            nums[i+1]=nums[j]
            j+= 1
            i+= 1
    return i+1

print(removeDuplicates(nums))
# removeDuplicates(nums)

File: Languages/Java/CodingChallenges/LeetCode/RemoveDuplicatesFromSortedArray/Attempt01.java
import java.util.Arrays;

// import jdk.javadoc.internal.doclets.formats.html.SourceToHTMLConverter;

public class Attempt01 {
    public static void main(String[] args) throws Exception {
        // int[] nums = { 1, 1, 2 };
        // int[] nums = {1, 1, 2, 3};
        int[] nums = {0, 0, 1, 1, 1, 2, 2, 3, 3, 4};
        System.out.print(removeDuplicates(nums));
    }

    // public int removeDuplicates(int[] nums) {
    static int removeDuplicates(int[] nums) {
        // System.out.print("\n-------------------------\n");
        // System.out.println("Original array: " + Arrays.toString(nums));
        // System.out.print("\n-------------------------\n");
        // System.out.print("Want to return length of 2 and [1, 2, 2]");
        // System.out.print("\n-------------------------\n");

        if (nums.length < 2) {
            return nums.length;
        }

        int i = 0;
        int j = 0;

        for (j = 0; j < nums.length-1;) {
            if (nums[i] == nums[j + 1]) {
                // System.out.print("They're equal. ");
                // System.out.print("Do this. ");
                j++;
            } else {
                // System.out.print("Do this instead.");
                nums[i+1]=nums[j+1];
                j++;
                i++;
            }
        }
        System.out.print("\n-------------------------\n");


        System.out.println("Modified array: " + Arrays.toString(nums));
        System.out.print("\n-------------------------\n");
        int length = i+1;

        return length;
        // }
    }
}